### PR TITLE
plugins.yupptv: support for yupptv with login support

### LINKED ISSUE #506: Add plugin: yupptv
I am suspecting that cookies/token expires and I am no longer able to play the stream, not even a restart as it returns **`403 Client Error: Forbidden`**

Example of the links I am getting:

`http://hlsbanglanews-lh.akamaihd.net/i/ams54_russiatoday@349571/master.m3u8?hdnea=st%3d1485750135%7eexp%3d1485750495%7eacl%3d%2fi%2fams54_russiatoday%40349571%2f*%7edata%3d1814225%7ehmac%3da3a3c4227a1c121aca77c38e9c21252bd881a5a9d5c96d7851ac8f53bb56c2a7&dw=160`

Streaming url: 

`http://hlsbanglanews-lh.akamaihd.net/i/ams54_russiatoday@349571/master.m3u8`
Token?:

`hdnea=st%3d1485750135%7eexp%3d1485750495%7eacl%3d%2fi%2fams54_russiatoday%40349571%2f*%7edata%3d1814225%7ehmac%3da3a3c4227a1c121aca77c38e9c21252bd881a5a9d5c96d7851ac8f53bb56c2a7&dw=160`

Seems like there is linux epoch time that sets the expiry of the stream. So each time i need to login to the website and request new token.

I also know the following:
SWF: `http://d3hprka3kr08q2.cloudfront.net/staticstorage/jwplayer/jw_7_8_2/jwplayer.loader.swf`

Content of M3U8 file: 

```
#EXTM3U
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=87000,RESOLUTION=256x144,CODECS="avc1.66.30, mp4a.40.2"
http://hlsbanglanews-lh.akamaihd.net/i/ams54_russiatoday@349571/index_64_av-p.m3u8?sd=10&dw=160&rebase=on&hdntl=exp=1485836971~acl=%2fi%2fams54_russiatoday%40349571%2f*~data=hdntl~hmac=a18a639e66837c960067f07eae81b129b3d20804fd69aaec598f138ab9321128
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=226000,RESOLUTION=384x216,CODECS="avc1.66.30, mp4a.40.2"
http://hlsbanglanews-lh.akamaihd.net/i/ams54_russiatoday@349571/index_200_av-p.m3u8?sd=10&dw=160&rebase=on&hdntl=exp=1485836971~acl=%2fi%2fams54_russiatoday%40349571%2f*~data=hdntl~hmac=a18a639e66837c960067f07eae81b129b3d20804fd69aaec598f138ab9321128
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=403000,RESOLUTION=480x270,CODECS="avc1.66.30, mp4a.40.2"
http://hlsbanglanews-lh.akamaihd.net/i/ams54_russiatoday@349571/index_400_av-p.m3u8?sd=10&dw=160&rebase=on&hdntl=exp=1485836971~acl=%2fi%2fams54_russiatoday%40349571%2f*~data=hdntl~hmac=a18a639e66837c960067f07eae81b129b3d20804fd69aaec598f138ab9321128
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=647000,RESOLUTION=512x288,CODECS="avc1.66.30, mp4a.40.2"
http://hlsbanglanews-lh.akamaihd.net/i/ams54_russiatoday@349571/index_800_av-p.m3u8?sd=10&dw=160&rebase=on&hdntl=exp=1485836971~acl=%2fi%2fams54_russiatoday%40349571%2f*~data=hdntl~hmac=a18a639e66837c960067f07eae81b129b3d20804fd69aaec598f138ab9321128
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=1233000,RESOLUTION=704x396,CODECS="avc1.66.30, mp4a.40.2"
http://hlsbanglanews-lh.akamaihd.net/i/ams54_russiatoday@349571/index_1100_av-p.m3u8?sd=10&dw=160&rebase=on&hdntl=exp=1485836971~acl=%2fi%2fams54_russiatoday%40349571%2f*~data=hdntl~hmac=a18a639e66837c960067f07eae81b129b3d20804fd69aaec598f138ab9321128
```

Also, few attributes such as "Referer" for the m3u8 stream.

Is it possible to somehow automate via streamlink the token request process? The website does require a log on in order to get the stream link.

### PR DESCRIPTION
**Pull request by beardypig**
I wasn't able to test subscription videos, but I assume it'll work.

Fixes #506

### CODE REVIEW

File: src/streamlink/plugins/yupptv.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            if error_code is None:
+                log.info("Logged in as {0}".format(self.get_option("email")))
+            else:
+                log.error("Failed to login: {1} (code: {0})".format(error_code, error_msg))
+
+        page = http.get(self.url)
+        match = self._m3u8_re.search(page.text)
+        if match:
+            stream_url = match.group(1)
+            if "preview/" in stream_url:
```
Comment (beardypig) [inline src/streamlink/plugins/yupptv.py]: This check might be a bit risky, but it is probably OK...

**Comment by gravyboat** [comment]
Thanks @beardypig.

**Comment by beardypig** [comment]
Never confirmed if this works with a subscription.

**Comment by gravyboat** [comment]
@beardypig Yeah I know but since no one commented in 10 days I said screw it. If people aren't going to test it out with their account after asking for it to be fixed there's nothing we can do, but we might as well merge it in.

**Comment by maanich** [comment]
yupp i sok with valid account tested and works stable   tried only live stream need to test vod once ok i will update. @beardypig

**Comment by maanich** [comment]
streamlink --yupptv-email xxxx@gmail.com --yupptv-password xxxxx --http-no-ssl-verify https://www.yuppflix.com/movies/watch/24-tamil  best
[cli][info] streamlink is running as root! Be careful!
error: No plugin can handle URL: https://www.yuppflix.com/movies/watch/24-tamil


its vod from yupp   live is fine as i stated in last post

**Comment by beardypig** [comment]
@maanich yuppflix is a different site.

**Pull request merged**